apps/dashboard/app/javascript/launcher_edit.js

```diff
@@ -166,16 +166,38 @@ function addInProgressField(event) {
 function updateAutoEnvironmentVariable(event) {
   var aev_name = event.target.value;
   const labelString = event.target.dataset.labelString;
+  const idString = `launcher_auto_environment_variable_${aev_name}`;
+  const nameString = `launcher[auto_environment_variable_${aev_name}]`;
   var input_field = event.target.parentElement.children[2].children[1];
 
   input_field.removeAttribute('readonly');
-  input_field.id = `launcher_auto_environment_variable_${aev_name}`;
-  input_field.name = `launcher[auto_environment_variable_${aev_name}]`;
+  input_field.id = idString;
+  input_field.name = nameString;
 
   if (labelString.match(/Environment(&#32;|\s)Variable/)) {
     var label_field = event.target.parentElement.children[2].children[0];
     label_field.innerHTML = `Environment Variable: ${aev_name}`;
   }
+
+  // Update the checkbox so that environment variables can be fixed when created
+  let fixedBoxGroup = event.target.parentElement.children[3].children[0].children[0];
```

Comment:
This looks like it's really flaky and prone to errors. Is there some ID or data- attribute we can apply here so we don't have this?

`querySelector` (or `querySelectorAll`) is a method on the `document` object, bit it's also a method on all elements. So you could do `event.target.querySelector` to find what we're looking for.